src/plugins/dashboard/public/application/utils/get_table_columns.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { History } from 'history';
+import { EuiLink } from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import { ApplicationStart } from 'opensearch-dashboards/public';
+import { IUiSettingsClient } from 'src/core/public';
+import moment from 'moment';
+
+export const getTableColumns = (
+  application: ApplicationStart,
+  history: History,
+  uiSettings: IUiSettingsClient
+) => {
+  const dateFormat = uiSettings.get('dateFormat');
+
+  return [
+    {
+      field: 'title',
+      name: i18n.translate('dashboard.listing.table.titleColumnName', {
+        defaultMessage: 'Title',
+      }),
+      sortable: true,
+      render: (field: string, record: { viewUrl?: string; title: string }) => (
+        <EuiLink
+          href={record.viewUrl}
+          data-test-subj={`dashboardListingTitleLink-${record.title.split(' ').join('-')}`}
```

Comment:
out of scope for this PR, but we should just make a helper function for properly formatting data test subject strings.